raphtory-graphql/src/model/graph/namespace.rs

```diff
@@ -0,0 +1,109 @@
+use crate::{
+    data::get_relative_path,
+    model::graph::meta_graph::MetaGraph,
+    paths::{valid_path, ExistingGraphFolder},
+};
+use dynamic_graphql::{ResolvedObject, ResolvedObjectFields};
+use std::path::PathBuf;
+use walkdir::WalkDir;
+
+#[derive(ResolvedObject)]
+pub(crate) struct Namespace {
+    base_dir: PathBuf,
+    current_dir: PathBuf,
```

Comment:
we would probably be better off making this `String` and storing the relative validated path here.